test/sanity/globalConstants.wtest

```diff
@@ -0,0 +1,103 @@
+const x = 1 + 1
+test "global constants are referenceable" {
+  assert.equals(2, x)
+}
+
+
+const a = b
+const b = 1 + 1
+const c = b
+test "global constants referencing other constants defined before/after are properly initialized" {
+  assert.equals(2, a)
+  assert.equals(2, b)
+  assert.equals(2, c)
+}
+
+const o = p
+object p {
+  const property e = self.f()
+  const property f = o
+  const property g = q
+  const property h = o.itself()
+  const property i = q.itself()
```

Comment:
If the singletons get instantiated first (without initializing them) we could solve `f` and `g` if the order for initialization was: `o`, `q`, `p.f`, `p.g`. I believe `h` and `i` could work, as long as itself() does not depend on p fields.